wweval/R/supplement_analyses_and_figs.R

```diff
@@ -1058,3 +1058,87 @@ get_heatmap_metadata_hub <- function(metadata,
     )
   )
 }
+
+#' Get the overall relative crps for the real time scores
+#'
+#' @param all_scores a tibble of the scores for both models in real-time
+#'
+#'
+#' @return A tibble summarized the relative crps across all locs
+#' and forecast dates produced in real-time
+#' @export
+get_overall_rel_crps_rt <- function(all_scores) {
+  full_metadata <- all_scores |>
+    dplyr::filter(scale == "log") |>
+    dplyr::group_by(forecast_date, model, location, failed_convergence) |>
+    dplyr::summarize(mean_crps = mean(crps)) |>
+    dplyr::filter(failed_convergence == FALSE) |>
+    tidyr::pivot_wider(
+      names_from = model,
+      values_from = mean_crps
+    )
+  # Find the date locations to exclude
+
+  date_locs_to_exclude <- full_metadata |>
+    dplyr::filter(is.na(ww)) |>
+    dplyr::distinct(location, forecast_date)
+
+  scores_filtered <- all_scores |>
+    dplyr::anti_join(date_locs_to_exclude,
+      by = c("location", "forecast_date")
+    )
+
+  rel_scores <- scores_filtered |>
+    dplyr::filter(scale == "log") |>
+    dplyr::select(location, forecast_date, date, model, crps) |>
+    tidyr::pivot_wider(
+      names_from = model,
+      values_from = crps
+    ) |>
+    dplyr::mutate(
+      rel_crps = ww / hosp
+    )
+
+  avg_rel_scores <- rel_scores |>
+    dplyr::group_by(forecast_date, location) |>
+    dplyr::summarise(mean_rel_crps = mean(rel_crps))
+
+  overall_rel_crps <- rel_scores |>
+    dplyr::ungroup() |>
+    dplyr::summarise(mean_rel_crps = mean(rel_crps))
+
+  p <- ggplot(avg_rel_scores) +
```

Comment:
Not ideal to have a function that computes/returns one thing and also saves a plot of another thing as a side effect. Fixing this can wait if this PR really needs to land, but if it's quick to fix we should.

Reply:
Yeah I can just break it up into two functions, it just bloats the targets pipeline (but I guess it is already very bloated). And don't want to introduce redundancy.